pkg/applicationserver/io/web/templates.go

```diff
@@ -165,6 +166,9 @@ func (ts *TemplateStore) allTemplateIDs() (ids []string, err error) {
 }
 
 func (ts *TemplateStore) getAllTemplateIDs() ([]string, error) {
+	if ts == nil {
```

Comment:
This feels weird to me; the caller should not call this method if the store is nil.

Reply:
The nil check is now done on the method implementation. It feels weird there as well, since that one is also under `TemplateStore`. Resolving this conversation, also see my comment below.

Reply:
Yes I see this is done on the method implementation, and that is not how it's supposed to be.

This would only work if we register a nil-service implementation. That would be wrong to do.

We would either register a "not found" service implementation, simply returning 404s on getters, or not register the service at all, leading to Unimplemented errors (probably 500s).

Up to @adriansmares to decide